src/Plugins/ComplexCore/src/ComplexCore/Filters/ImportBinaryCTNorthstarFilter.cpp

```diff
@@ -0,0 +1,551 @@
+                                                                 dataFilePair.first.string(), pInputHeaderFileValue.parent_path().string()))};
+    }
+  }
+
+  s_HeaderCache[m_InstanceId].DataFilePaths = dataFiles;
+
+  // Set the output actions
+  complex::Result<OutputActions> resultOutputActions;
+  auto createImageGeometryAction =
+      std::make_unique<CreateImageGeometryAction>(pImageGeometryPathValue, importedGeometryInfo.Dimensions, importedGeometryInfo.Origin, importedGeometryInfo.Spacing, pCellAttributeMatrixNameValue);
+  resultOutputActions.value().actions.push_back(std::move(createImageGeometryAction));
+
+  DataPath densityArrayPath = pImageGeometryPathValue.createChildPath(pCellAttributeMatrixNameValue).createChildPath(pDensityArrayNameValue);
+  //  std::vector<usize> tupleDims = {importedGeometryInfo.Dimensions[2], importedGeometryInfo.Dimensions[1], importedGeometryInfo.Dimensions[0]};
+  resultOutputActions.value().actions.push_back(std::make_unique<CreateArrayAction>(DataType::float32, importedGeometryInfo.Dimensions, std::vector<usize>{1}, densityArrayPath));
+
+  // Set the preflight updated values
+  std::string volumeDescription = GenerateGeometryInfoString(geometryInfo, static_cast<IGeometry::LengthUnit>(pLengthUnitValue));
+  std::string dataFileInfo = GenerateDataFileListInfoString(dataFiles);
+
+  std::vector<PreflightValue> preflightUpdatedValues;
+  preflightUpdatedValues.push_back({"Original Volume", volumeDescription});
+  preflightUpdatedValues.push_back({"Data Files", dataFileInfo});
+
+  if(pImportSubvolumeValue)
+  {
+    std::string importedVolumeDescription = GenerateGeometryInfoString(importedGeometryInfo, static_cast<IGeometry::LengthUnit>(pLengthUnitValue));
+    preflightUpdatedValues.push_back({"Subvolume", importedVolumeDescription});
+  }
+
+  // Return output actions and preflight updated values
+  return {std::move(resultOutputActions), std::move(preflightUpdatedValues)};
+}
+
+//------------------------------------------------------------------------------
+Result<> ImportBinaryCTNorthstarFilter::executeImpl(DataStructure& dataStructure, const Arguments& filterArgs, const PipelineFilter* pipelineNode, const MessageHandler& messageHandler,
+                                                    const std::atomic_bool& shouldCancel) const
+{
+  ImportBinaryCTNorthstarInputValues inputValues;
+
+  inputValues.InputHeaderFile = filterArgs.value<FileSystemPathParameter::ValueType>(k_InputHeaderFile_Key);
+
+  auto pImageGeometryPathValue = filterArgs.value<DataPath>(k_ImageGeometryPath_Key);
+  inputValues.ImageGeometryPath = pImageGeometryPathValue;
+
+  auto pCellAttributeMatrixNameValue = filterArgs.value<std::string>(k_CellAttributeMatrixName_Key);
+  auto pDensityArrayNameValue = filterArgs.value<std::string>(k_DensityArrayName_Key);
+  DataPath densityArrayPath = pImageGeometryPathValue.createChildPath(pCellAttributeMatrixNameValue).createChildPath(pDensityArrayNameValue);
```

Comment:
```suggestion
  inputValues.ImageGeometryPath = filterArgs.value<DataPath>(k_ImageGeometryPath_Key);

  auto pCellAttributeMatrixNameValue = filterArgs.value<std::string>(k_CellAttributeMatrixName_Key);
  auto pDensityArrayNameValue = filterArgs.value<std::string>(k_DensityArrayName_Key);
  DataPath densityArrayPath = inputValues.ImageGeometryPath.createChildPath(pCellAttributeMatrixNameValue).createChildPath(pDensityArrayNameValue);
```
Avoid the unnecessary copy if possible